### ISSUE TITLE
Navs: nav expansion on GJ QSRE in projection doesn't introduce parameter

### ISSUE THREAD

**Issue by anpete** [aspnet/EntityFramework#8186]
```c#
from c in context.Set<Customer>()
join o in context.Set<Order>()
on c.CustomerID equals o.CustomerID into g
where c.CustomerID == "ALFKI"
select new { c, G = g.Select(o => _Include(o, o.OrderDetails)).ToList() })
```
produces:
```sql
SELECT [c].[CustomerID], [c].[Address], [c].[City], [c].[CompanyName], [c].[ContactName], [c].[ContactTitle], [c].[Country], [c].[Fax], [c].[Phone], [c].[PostalCode], [c].[Region], [o].[OrderID], [o].[CustomerID], [o].[EmployeeID], [o].[OrderDate]
FROM [Customers] AS [c]
LEFT JOIN [Orders] AS [o] ON [c].[CustomerID] = [o].[CustomerID]
WHERE [c].[CustomerID] = N'ALFKI'
ORDER BY [c].[CustomerID]
  
SELECT [o0].[OrderID], [o0].[ProductID], [o0].[Discount], [o0].[Quantity], [o0].[UnitPrice]
FROM [Order Details] AS [o0]

SELECT [o0].[OrderID], [o0].[ProductID], [o0].[Discount], [o0].[Quantity], [o0].[UnitPrice]
FROM [Order Details] AS [o0]
```
```c#
from c in context.Set<Customer>()
join o in context.Set<Order>()
on c.CustomerID equals o.CustomerID into g
where c.CustomerID == "ALFKI"
select new { c, G = g.Select(o => _Include(o, o.Customer)).ToList() }
```
produces:
```sql
SELECT [c].[CustomerID], [c].[Address], [c].[City], [c].[CompanyName], [c].[ContactName], [c].[ContactTitle], [c].[Country], [c].[Fax], [c].[Phone], [c].[PostalCode], [c].[Region], [o].[OrderID], [o].[CustomerID],
[o].[EmployeeID], [o].[OrderDate]
FROM [Customers] AS [c]
LEFT JOIN [Orders] AS [o] ON [c].[CustomerID] = [o].[CustomerID]
WHERE [c].[CustomerID] = N'ALFKI'
ORDER BY [c].[CustomerID]

SELECT [o.Customer].[CustomerID], [o.Customer].[Address], [o.Customer].[City], [o.Customer].[CompanyName], [o.Customer].[ContactName], [o.Customer].[ContactTitle], [o.Customer].[Country], [o.Customer].[Fax], [o.Customer].[Phone], [o.Customer].[PostalCode], [o.Customer].[Region]
FROM [Customers] AS [o.Customer]

```

**Comment by smitpatel** [comment]
Quick summary for @anpete of the discussion.

Binding to outer parameter is relational concept at present. That means we bind to outer parameter only if QSRE on which outer parameter is generated is fully translated.
E.g. `q.property` will be converted to outer parameter only if `q` is result of SelectExpression or part of it. (but outer parameter because we couldn't lift current QM to outer QM.) It is that way because the logic is we find SE for querysource and then bind the property by using projection.

For this case, the outer QM is not exactly SelectExpression since there is client side processing for group join. So when we try to find the SelectExpression for querysource, there is none and we cannot convert to outer parameter. Client eval follows.

What should happen is, if we cannot find SelectExpression then we should fall to Core Level outer parameter introduction which basically binds with result of client _GroupJoin. At present we already have way to translate this (which is called when we do client translation of predicate but it should be applied to left/right parts of predicate separately). The translation generated for client predicate's left side can be used for outer parameter. Introduction of this will ensure that whenever we are producing N+1 queries we always introduce outer parameter because we can generate outer parameter with client side methods too.

**Comment by maumar** [comment]
fixed in 32a196ac3b8f6394404f81b774fd82091d4708fa
